tools/jarcat/tar/tar_test.go

```diff
@@ -0,0 +1,105 @@
+package tar
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const testDataDir = "tools/jarcat/tar/test_data"
```

Comment:
could the outputs be written in-memory instead, e.g with `bytes.Buffer` ? Or is that an issue because of filepath.Walk? If so, you could use `ioutil.Temp*`, which would already be in-memory for modern OSes